S7.Net/PlcAsynchronous.cs

```diff
@@ -54,15 +78,12 @@ public async Task OpenAsync(CancellationToken cancellationToken = default)
             if (s7data.Length < 20)
                 throw new WrongNumberOfBytesException("Not enough data received in response to Communication Setup");
 
-            MaxPDUSize = (short)(s7data[18] * 256 + s7data[19]);
-        }
-
-        private async Task ConnectAsync()
-        {
-            tcpClient = new TcpClient();
-            ConfigureConnection();
-            await tcpClient.ConnectAsync(IP, Port);
-            _stream = tcpClient.GetStream();
+            // TODO: check if this should not rather be UInt16.
+            MaxPDUSize = Types.Int.FromByteArray(s7data, 18);
```

Comment:
Which also makes this invalid (but possibly the next byte is always 0, or at least always 0 in response to _our_ ConnectionRequest).